### ISSUE TITLE
second stage loss

### ISSUE THREAD

**Issue by saudades18** [SamsungLabs/NeuralHaircut#12]
hi, can someone help me to solve this error?
i try to train the second stage using the data you provide(because the memory limitation, i use 40 images ),  and the loss isn't decreasing.

![image](https://github.com/SamsungLabs/NeuralHaircut/assets/143250003/5f9c0d85-8795-42eb-afba-2b58d4e31960)

![image](https://github.com/SamsungLabs/NeuralHaircut/assets/143250003/d7672dd2-c0ae-468e-872a-f1b43157fac1)

**Comment by VanessaSklyarova** [comment]
Hi @saudades18! Could you please describe how you launch the code on 40 images?

**Comment by saudades18** [comment]
thanks for your reply @VanessaSklyarova .i just change the monoculardataset and add [:40] at the end of string in 330-340

**Comment by saudades18** [comment]
And i use h3ds dataset to test, howerer, during the training, it still occur "NaN during backprop was found, skipping iteration", is it normal?

**Comment by VanessaSklyarova** [comment]
@saudades18 did you face any uploading weights error in https://github.com/SamsungLabs/NeuralHaircut/blob/main/src/models/dataset.py#L378 and did you use configs from ./examples? 

Yes, "NaN during backprop was found, skipping iteration" is okay if it happens not frequently.

**Comment by saudades18** [comment]
@VanessaSklyarova i did not face uploading weights error, however  "NaN during backprop was found, skipping iteration" happens frequently, about 90%.  
And when i using h3ds dataset, the hair is optimized. but when using the data given(this time all images), the loss still was not decreased, and the hair primitives were still long and straight.
Could you provide the checkpoint for the second stage?
Thanks for your reply :)

**Comment by VanessaSklyarova** [comment]
@saudades18  Could you set https://github.com/SamsungLabs/NeuralHaircut/blob/main/configs/example_config/hair_strands_textured.yaml#L54 to false, change https://github.com/SamsungLabs/NeuralHaircut/blob/main/configs/example_config/hair_strands_textured.yaml#L77 and https://github.com/SamsungLabs/NeuralHaircut/blob/main/configs/example_config/hair_strands_textured.yaml#L78 to 0. and check if the loss decrease with time?

**Comment by saudades18** [comment]
@VanessaSklyarova  Thanks for your kind reply! Now the hair is optimized, and the loss is decreasing except hair_L_diff loss. And before most of the render images are black, maybe this is the reason? So how can i train successfully with rendering?

**Comment by VanessaSklyarova** [comment]
@saudades18 Yes, it should be more stable if you start rendering after approx. 1000 steps,  but still it is very strange why it doesn't work from the beginning. I didn't face such rendering problems before when checked on different scenes, so I'll have a look at this.

**Comment by saudades18** [comment]
@VanessaSklyarova i started rendering after 1000 steps but it seems render images still not be optimized, and the "NaN" still happened frequently. Below is one of the render images.
![pred_hair_strands_7](https://github.com/SamsungLabs/NeuralHaircut/assets/143250003/c9ae4dd4-428a-422b-8b69-80c91e693601)

**Comment by saudades18** [comment]
@VanessaSklyarova i start rendering after 1000 steps, but it seems like the render images are still not optimized. And "Nan" still happens frequently. Below is one of the render images, and it looks like rasterize image rather than real rgb.
![pred_hair_strands_6](https://github.com/SamsungLabs/NeuralHaircut/assets/143250003/5a94bb97-3216-458e-bfeb-155a0e5ab947)
